crates/sui-graphql-rpc/src/types/dynamic_field.rs

```diff
@@ -172,66 +164,126 @@ impl DynamicField {
         };
 
         let field_id = derive_dynamic_field_id(parent, &type_, &name.bcs.0)
-            .map_err(|e| Error::Internal(format!("Failed to derive dynamic field id: {e}")))?
-            .to_vec();
+            .map_err(|e| Error::Internal(format!("Failed to derive dynamic field id: {e}")))?;
+
+        let key = match parent_version {
+            Some(version) => ObjectLookupKey::VersionAt {
+                version,
+                checkpoint_viewed_at,
+            },
+            None => match checkpoint_viewed_at {
```

Comment:
IIRC, this is going to query for the dynamic field *at* the parent version (an equality check).  This is not correct -- it needs to query for the dynamic field with the highest version that is less than or equal to the parent version (bounds check).

Reply:
Thanks @amnn  this finally clicked for me :sweat_smile: